### PR TITLE
Images were named wrong and GITHUBURL env variable was not updated

### PR DESCRIPTION
**Pull request by betatim**
Solution: Skip repo url caching and always update env.

### CODE REVIEW

File: everware/spawner.py [hunk 1]
```diff
@@ -134,15 +134,13 @@ def options_from_form(self, formdata):
     def repo_url(self):
         return self.user_options['repo_url']
 
-    _escaped_repo_url = None
```
Comment (ibab) [inline everware/spawner.py:143]: Maybe we can get rid of `_escaped_repo_url` completely?

**Comment by ibab** [comment]
Had some problems verifying that the patch solves the problem, but after reinstall jupyterhub and the dockerspawner, it's working well for me.
I'm happy to merge this once you remove the `_escaped_repo_url`!

**Comment by betatim** [comment]
How would we ditch it though? It is used to derive the image name :confused:

File: everware/spawner.py [hunk 2]
```diff
@@ -134,15 +134,13 @@ def options_from_form(self, formdata):
...
-            trans = str.maketrans(':/-.', "____")
-            repo_url = self.repo_url.translate(trans).lower()
-            if repo_url.endswith('.git'):
-                repo_url = repo_url[:-4]
-            self._escaped_repo_url = re.sub("_+", "_", repo_url)
+        trans = str.maketrans(':/-.', "____")
+        repo_url = self.repo_url.translate(trans).lower()
+        if repo_url.endswith('.git'):
+            repo_url = repo_url[:-4]
+        self._escaped_repo_url = re.sub("_+", "_", repo_url)
```
Comment (astiunov) [inline everware/spawner.py]: We don't need self._escaped_repo_url

**Comment by betatim** [comment]
This is buggy, I'll add a test and fix the bug.

**Comment by betatim** [comment]
Does someone understand why this fails? On my machine it runs to completion. From the info in travis' log I can't really work it out :(

**Comment by betatim** [comment]
Can reproduce this locally now. Started with setting up everware from scratch.

Problem: The container isn't passed a sensible URL to clone from

**Comment by betatim** [comment]
Works with `dockerspawner` `master` but not the latest released version. Released version uses `self.env` and `master` uses `self.get_env()`

See jupyter/dockerspawner#71

Should we update our dependency to a commit to `master`? Or maybe pin it to a certain commit after the release.

**Comment by betatim** [comment]
- [ ] add a test that fails with the old code and passes after these changes

**Pull request merged**